Remove IContainerRuntime.flush

kian-thompson opened the pull request:
[AB#1076](https://dev.azure.com/fluidframework/internal/_workitems/edit/1076)

For more information about how to contribute to this repo, visit [this page](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md).

## Description

`IContainerRuntime.flush` was deprecated in a previous version.
It's removal introduces a breaking change. If a more manual flush functionality is needed, migrate all usage to `IContainerRuntimeBase.orderSequentially` if possible.

kian-thompson committed:
Remove IContainerRuntime.flush

andre4i commented on BREAKING.md:
```diff
@@ -36,9 +36,14 @@ This field has been deprecated and will be removed in a future breaking change.
         fileName: string,
 -       createShareLinkType?: ShareLinkTypes,
 +       createShareLinkType?: ShareLinkTypes | ISharingLinkKind,
```
This is the second time today that I see this change in BREAKING.md. Is your repo out of sync?

andre4i commented on BREAKING.md:
````diff
@@ -36,9 +36,14 @@ This field has been deprecated and will be removed in a future breaking change.
 ```
 
+## 3.0.0 Breaking changes
+- [Remove `IContainerRuntime.flush`](#icontainerruntimeflush-is-deprecated)
````
This link won't work.

andre4i commented on packages/test/local-server-tests/src/test/documentDirty.spec.ts:
```diff
@@ -319,9 +319,6 @@ describe("Document Dirty", () => {
 
                     checkDirtyState("after batch value set", true, 0);
 
-                    // Manually flush the ops so that they are sent as a batch.
```
these should probably be changed to ordersequentially, I'm not sure if 'immediate' as flushmode would work

kian-thompson committed:
Fix BREAKIN.md

kian-thompson committed:
Fix unit tests
- batch metadata is now added regardless if the container is connected (needs this information when replaying Immediate batches)

kian-thompson committed:
Attempt to fix unit tests

kian-thompson committed:
Merge branch 'next' into 1076-remove-flush

kian-thompson committed:
Fix some unit tests

kian-thompson committed:
Merge branch 'next' into 1076-remove-flush

kian-thompson committed:
Fix BREAKING.md

kian-thompson commented on BREAKING.md:
```diff
@@ -58,19 +58,23 @@ This field has been deprecated and will be removed in a future breaking change.
 `ISummaryConfigurationHeuristics.idleTime` has been removed. See [#10008](https://github.com/microsoft/FluidFramework/issues/10008)
 Please move all usage to the new `minIdleTime` and `maxIdleTime` properties in `ISummaryConfigurationHeuristics`.
 
-### Move `TelemetryNullLogger` and `BaseTelemetryNullLogger` to telemetry-utils package
```
I didn't catch the weird ordering in my last PR into next. These moved items belong under the `upcoming changes` section

kian-thompson committed:
Simple commit to trigger new build

kian-thompson committed:
Merge branch 'next' into 1076-remove-flush
